### PR TITLE
feature/TSS-889/add-new-search-fields-included-and-changedfor-mve

### PR DESCRIPTION
**Pull request by osimuka**

### CODE REVIEW

File: api/barriers/models.py [hunk 1]
```diff
@@ -1390,7 +1393,9 @@ def sector_filter(self, queryset, name, value):
         custom filter for multi-select filtering of Sectors field,
         which is ArrayField
         """
-        return queryset.filter(Q(all_sectors=True) | Q(sectors__overlap=value))
+        return queryset.filter(Q(main_sector__in=value)).filter(
+            Q(all_sectors=True) | Q(sectors__overlap=value)
+        )
```
Comment (Santino-Trade) [inline api/barriers/models.py:1398]: We'll need a unit test for this - This seems to be one of those Ticket vs Design situations - from what i understand, we need a tickbox to indicate that we only want the search term to apply to main_sector, or if we want the search term to apply for main_sector AND sector - is that the behaviour this filter is going to replicate?

Comment (osimuka) [reply api/barriers/models.py:1398]: @Santino-Trade I had a discussion with @ferozerub prior, that field is no longer required

File: tests/barriers/test_list_barriers.py [hunk 2]
```diff
@@ -499,6 +499,45 @@ def test_list_barriers_text_filter_based_on_summary(self):
...
+        barrier.save()
+
+        barrier2 = BarrierFactory()
+
+        barrier2.companies = [
+            {"id": "10", "name": "Springfield Nuclear Power Plant LTD"},
+        ]
+        barrier2.save()
+
+        url = f'{reverse("list-barriers")}?search=test company 3'
```
Comment (Santino-Trade) [inline tests/barriers/test_list_barriers.py:534]: Linked to the comment above, can we do a unit test here to test for the "Agent" company, to show the search applies for any company in a longer list of companies

File: api/barriers/models.py [hunk 3]
```diff
@@ -1553,11 +1558,15 @@ def text_search(self, queryset, name, value):
             full text search on summary
             partial search on title
         """
-        return queryset.annotate(search=SearchVector("summary"),).filter(
+        return queryset.annotate(
+            search=SearchVector("summary", "export_description"),
+            company_name=RawSQL("(companies->0->>'name')", ()),
```
Comment (Santino-Trade) [inline api/barriers/models.py:1563]: Next catchup, can you walk us through "(companies->0->>'name')" ? is this looping every company in the companies column? Turning the column into a single string?

**Review by ferozerub — APPROVED** [review]

**Pull request merged**